src/main/java/core/basesyntax/serviceimpl/DataParserImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataParserImpl implements DataParser {
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+    private static final String REGEX_FOR_SPLIT = ",";
+
+    @Override
+    public List<FruitTransaction> parse(List<String> dataFromFile) {
+        if (dataFromFile.isEmpty()) {

```

Comment:
Avoid throwing raw runtime exceptions like `IllegalArgumentException`. It's better to define your own exception class that provides more context about the error.